src/bp/bench/index.ts

```diff
@@ -0,0 +1,169 @@
+import axios from 'axios'
+import _ from 'lodash'
+import moment from 'moment'
+import ms from 'ms'
+import nanoid from 'nanoid'
+
+import { Stats } from './stats'
+
+export default class Bench {
+  stats: Stats
+  /** The default payload that each users will send */
+  defaultMessage: any
+  /** The target percentage of messages which must be under the limit sla */
+  slaTarget: number = 100
+  /** Messages taking more than this time are not meeting the desired SLA */
+  slaLimit: number = 1500
+  /** Number of users sending a message simultaneously */
+  users: number = 10
+  /** Number of messages that each users will send, one after another */
+  messages: number = 5
+  /**
+   * When this value is greater than 0, the benchmark scenario will start over and increment the number of users by this value.
+   * The scenario will be repeated until the speficied SLA is no longer respected
+   */
+  increments: number = 0
+  /** The base URL of the bot, including the port number, if any */
+  url: string
+  botId: string
+  /** When enabled, each message sent will be displayed with their response time */
+  isVerbose: boolean
+  interval: NodeJS.Timeout | undefined
+
+  constructor(args) {
+    this.url = args.url
+    this.botId = args.botId
+    this.slaTarget = args.slaTarget
+    this.slaLimit = args.slaLimit
+    this.users = args.users
+    this.messages = args.messages
+    this.increments = args.increments
+    this.isVerbose = Boolean(args.verbose)
+
+    this.stats = new Stats(this.slaLimit, this.slaTarget)
+    this.defaultMessage = {
+      type: 'text',
+      text: 'Hey!'
```

Comment:
I would make the text configurable as well so people can implement a special flow for load testing

Reply:
good idea